File: job-agent-os/src/job_agent_os/schemas/resume.py
"""Resume schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class EducationInfo(BaseModel):
    """Education information."""

    school: str | None = None
    degree: str | None = None
    major: str | None = None
    graduation: str | None = None


class ProjectExp(BaseModel):
    """Project experience."""

    name: str
    description: str | None = None
    tech_stack: list[str] = Field(default_factory=list)


class InternExp(BaseModel):
    """Internship experience."""

    company: str
    role: str | None = None
    duration: str | None = None
    description: str | None = None


class StructuredResumeData(BaseModel):
    """Structured resume data."""

    name: str | None = None
    education: EducationInfo | None = None
    skills: list[str] = Field(default_factory=list)
    projects: list[ProjectExp] = Field(default_factory=list)
    internships: list[InternExp] = Field(default_factory=list)
    certifications: list[str] = Field(default_factory=list)


class ResumeCreate(BaseModel):
    """Resume creation metadata."""

    title: str = Field(max_length=200)
    target_direction: str | None = None


class ResumeResponse(BaseModel):
    """Resume response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    version: str
    file_type: str | None = None
    file_path: str | None = None
    file_size_bytes: int | None = None
    is_encrypted: bool = False
    structured_data: dict = Field(default_factory=dict)
    target_direction: str | None = None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class ResumeUpdate(BaseModel):
    """Resume update request."""

    title: str | None = None
    structured_data: dict | None = None
    is_active: bool | None = None

File: job-agent-os/frontend/src/types/resume.ts
/** Resume types (matches backend schemas/resume.py) */

export interface EducationInfo {
  school: string | null;
  degree: string | null;
  major: string | null;
  graduation: string | null;
}

export interface ProjectExp {
  name: string;
  description: string | null;
  tech_stack: string[];
}

export interface InternExp {
  company: string;
  role: string | null;
  duration: string | null;
  description: string | null;
}

export interface StructuredResumeData {
  name?: string | null;
  education?: EducationInfo | null;
  skills?: string[];
  projects?: ProjectExp[];
  internships?: InternExp[];
  certifications?: string[];
}

export interface ResumeResponse {
  id: string;
  title: string;
  version: string;
  file_type: string | null;
  structured_data: StructuredResumeData | Record<string, unknown>;
  target_direction: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ResumeUpdate {
  title?: string | null;
  structured_data?: Record<string, unknown> | null;
  is_active?: boolean | null;
}
